diff --git a/./meta-openembedded/meta-oe/recipes-devtools/php/php.inc b/./php.inc
index 5345f2a..9b48984 100644
--- meta-openembedded/meta-oe/recipes-devtools/php/php.inc
+++ meta-openembedded/meta-oe/recipes-devtools/php/php.inc
@@ -9,7 +9,7 @@ DEPENDS_virtclass-native = "zlib-native libxml2-native"

 INC_PR = "r5"

-SRC_URI = "http://www.php.net/distributions/php-${PV}.tar.bz2"
+SRC_URI = "http://museum.php.net/php5/php-${PV}.tar.bz2"

 S = "${WORKDIR}/php-${PV}"

