--- eog-2.28.2/src/eog-uri-converter.c
+++ eog-2.28.2/src/eog-uri-converter.c
@@ -197,7 +197,7 @@
 		break;
 
 	case PROP_SPACE_CHARACTER:
-		priv->space_character = g_value_get_char (value);
+		priv->space_character = g_value_get_schar (value);
 		break;
 
 	case PROP_COUNTER_START:
@@ -249,7 +249,7 @@
 		break;
 
 	case PROP_SPACE_CHARACTER:
-		g_value_set_char (value, priv->space_character);
+		g_value_set_schar (value, priv->space_character);
 		break;
 
 	case PROP_COUNTER_START:
