diff --git a/src/events.c b/src/events.c
index e6cf0ac..66db3b1 100644
--- a/src/events.c
+++ b/src/events.c
@@ -101,6 +101,7 @@ void init_buttonbindings(void)
 	memset(&buttons, 0, sizeof(buttons));
 
 	feh_set_bb(&buttons.reload, 0, 0);
+	feh_set_bb(&buttons.quit,   0, 0);
 	feh_set_bb(&buttons.pan,    0, 1);
 	feh_set_bb(&buttons.zoom,   0, 2);
 	feh_set_bb(&buttons.menu,   0, 3);
@@ -150,6 +151,8 @@ void init_buttonbindings(void)
 			cur_bb = &buttons.next;
 		else if (!strcmp(action, "blur"))
 			cur_bb = &buttons.blur;
+		else if (!strcmp(action, "quit"))
+			cur_bb = &buttons.quit;
 		else if (!strcmp(action, "rotate"))
 			cur_bb = &buttons.rotate;
 		else if (!strcmp(action, "zoom_in"))
@@ -320,6 +323,9 @@ static void feh_event_handle_ButtonPress(XEvent * ev)
 		if (winwid->type == WIN_TYPE_SLIDESHOW)
 			slideshow_change_image(winwid, SLIDE_NEXT, 1);
 
+	} else if (feh_is_bb(&buttons.quit, button, state)) {
+		winwidget_destroy_all();
+
 	} else {
 		D(("Received other ButtonPress event\n"));
 	}
diff --git a/src/options.h b/src/options.h
index 27d3d38..7b18853 100644
--- a/src/options.h
+++ b/src/options.h
@@ -209,6 +209,7 @@ struct __fehbb {
 	struct __fehbutton rotate;
 	struct __fehbutton zoom_in;
 	struct __fehbutton zoom_out;
+	struct __fehbutton quit;
 };
 
 void init_parse_options(int argc, char **argv);
