Add cfa10036 support

Signed-off-by: Alexandre Belloni <alexandre.belloni@free-electrons.com>

diff --git a/Makefile b/Makefile
index 367738c..a9c5d21 100644
--- a/Makefile
+++ b/Makefile
@@ -16,6 +16,9 @@ endif
 ifeq ($(BOARD), iMX28_EVK)
 ARCH = mx28
 endif
+ifeq ($(BOARD), cfa10036)
+ARCH = mx28
+endif
 
 all: build_prep gen_bootstream
 
diff --git a/boot_prep/Makefile b/boot_prep/Makefile
index 0267ede..86570fc 100644
--- a/boot_prep/Makefile
+++ b/boot_prep/Makefile
@@ -23,6 +23,30 @@ MEM_TYPE ?= MEM_DDR1
 CFLAGS 	= -g -Wall -I$(INCLUDEDIR) -I$(INCLUDEDIR)/mach -I./ -O -D$(MEM_TYPE)
 LDFLAGS = -static -nostdlib -T $(BOOT_LAYOUT)
 
+# The board defines STMP CPU family and peripherial components
+# So, set ARCH and HW_OBJS variables for each supported board
+# separately.
+ifeq ($(BOARD), stmp37xx_dev)
+ARCH = 37xx
+HW_OBJS = $(LRADC_OBJS)
+CFLAGS += -DSTMP37XX -DBOARD_STMP37XX_DEV
+endif
+ifeq ($(BOARD), stmp378x_dev)
+ARCH = mx23
+HW_OBJS = $(LRADC_OBJS)
+CFLAGS += -DSTMP378X -DBOARD_STMP378X_DEV
+endif
+ifeq ($(BOARD), iMX28_EVK)
+ARCH = mx28
+HW_OBJS = $(LRADC_OBJS)
+CFLAGS += -DMX28 -DBOARD_MX28_EVK
+endif
+ifeq ($(BOARD), cfa10036)
+ARCH = mx28
+HW_OBJS = $(LRADC_OBJS)
+CFLAGS += -DMX28 -DBOARD_CFA10036
+endif
+
 # Generic code
 CORE_OBJS = init-$(ARCH).o debug.o
 
